allennlp/data/dataset_readers/textcat.py

```diff
@@ -0,0 +1,129 @@
+from typing import Dict, List
+import logging
+import json
+import numpy as np
+from overrides import overrides
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import LabelField, TextField, Field, ListField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.data.tokenizers import Tokenizer, WordTokenizer
+from allennlp.data.tokenizers.sentence_splitter import SpacySentenceSplitter
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("textcat")
+class TextCatReader(DatasetReader):
```

Comment:
I'm not sure how I feel about `TextCat` as a name for this - we like to avoid abbreviations as much as possible.  Is "textcat" a standard name somewhere?  In the other class names in this PR you use full names ("Classifier"); why not use a full name here, too?  For the class name and the name you register it with.

Reply:
I'd make the name even more verbose and add the JSON file format to the name, to easily distinguish from e.g. a `TextClassificationCSVReader`.